purchase_work_acceptance/models/work_acceptance.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class WorkAcceptance(models.Model):
+    _name = 'work.acceptance'
+    _description = 'Work Acceptance'
+    _order = 'id desc'
+
+    name = fields.Char(
+        required=True,
+        index=True,
+        copy=False,
+        default='New'
+    )
+    origin = fields.Char(
+        string='Source Document',
+        copy=False,
+        help="Reference of the document that generated this work acceptance."
+    )
+    date_due = fields.Datetime(
+        string='Due Date',
+        required=True,
+        copy=False,
+    )
+    date_receive = fields.Datetime(
+        string='Received Date',
+        required=True,
+        copy=False,
```

Comment:
If you have both required=True, and copy=False, you will get error immediatelly when copy.
```suggestion
```